### PR TITLE
G4 t2 format input

### PR DESCRIPTION
**Pull request by BuzatuCos**

### CODE REVIEW

File: src/FormDataHandler.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private $ingredientsArray=[];
+
+    /**
+     * @param $formData post from the form
+     * @return array of ingredients from the user , validated
+     */
+    public function processData(array $formData) :array{
+       foreach($formData as $key=>$value){
+           if($key == 'userIngredients'){
+               $userIngredients = explode (' ',$value);
```
Comment (sudokufan) [inline src/FormDataHandler.php]:
```suggestion
               $userIngredients = explode (' ', $value);
```

Comment (sudokufan) [reply src/FormDataHandler.php]: whitespace

File: src/FormDataHandler.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+               $userIngredients= UserIngredientValidator::validateUserIngredients($userIngredients);
+               $this->ingredientsArray=array_merge($this->ingredientsArray, $userIngredients);
+           }else{
```
Comment (sudokufan) [inline src/FormDataHandler.php]:
```suggestion
           } else {
```

File: src/FormDataHandler.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+class FormDataHandler
+{
+    /**
+     * @var array Property to hold validate ingredients
+     */
+    private $ingredientsArray=[];
+
+    /**
+     * @param $formData post from the form
+     * @return array of ingredients from the user , validated
```
Comment (sudokufan) [inline src/FormDataHandler.php]:
```suggestion
     * @return array of ingredients from the user, validated
```

**Review by sudokufan — COMMENTED** [review]
found some spots for improved readability, made suggestions

File: tests/src/FormDataHandler.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+    /**
+//     * Tests the type-hinting of the FormDataHandler function
+//     */
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php]:
```suggestion
```

File: tests/src/FormDataHandler.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    public function testFormDataHandlerFailure() {
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+    /**
+//     * Tests the type-hinting of the FormDataHandler function
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php]:
```suggestion
//  Tests the type-hinting of the FormDataHandler function
```

File: tests/src/FormDataHandler.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+    /**
+//  Tests the type-hinting of the FormDataHandler function
+    public function testFormDataHandlerMalformed()
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php:32]:
```suggestion
    public function testFormDataHandlerMalformed() */
```

File: tests/src/FormDataHandler.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    public function testFormDataHandlerFailure() {
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+    /**
+//  Tests the type-hinting of the FormDataHandler function
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php:31]:
```suggestion
/*  Tests the type-hinting of the FormDataHandler function
```

File: tests/src/FormDataHandler.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    public function testFormDataHandlerFailure() {
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+    /**
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php]:
```suggestion
```

File: tests/src/FormDataHandler.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    public function testFormDataHandlerFailure() {
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+/*  Tests the type-hinting of the FormDataHandler function
+    public function testFormDataHandlerMalformed() */
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php]:
```suggestion
    public function testFormDataHandlerMalformed()
```

File: tests/src/FormDataHandler.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public function testFormDataHandlerFailure() {
+        $formData = [10, 41, 3259703498,'345'];
+        $formDataHandler= new GRUB\FormDataHandler();
+        $validation  = $formDataHandler->processData($formData);
+        $expected = ['', '1', '2','3'];
+        $this->assertEquals($validation, $expected);
+    }
+
+/*  Tests the type-hinting of the FormDataHandler function
```
Comment (jdselby24) [inline tests/src/FormDataHandler.php]:
```suggestion
//  Tests the type-hinting of the FormDataHandler function
```

**Review by sudokufan — APPROVED** [review]
👍🏻🎉💃🏻🎊

**Pull request merged**